client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx

```diff
@@ -204,7 +213,23 @@ const GoalsStep: StepType = ( { navigation, flow } ) => {
 				<Step.CenteredColumnLayout
 					columnWidth={ 6 }
 					className="step-container-v2--goals"
-					topBar={ <Step.TopBar skipButton={ <Step.SkipButton onClick={ handleSkip } /> } /> }
+					topBar={
+						<Step.TopBar
+							skipButton={ <Step.SkipButton onClick={ handleSkip } /> }
+							backButton={
+								site?.plan?.is_free ? (
+									<Step.BackButton
+										onClick={ () => {
```

Comment:
Why does the back button need to handle where to go manually? Should the framework handle this internally?

Reply:
Because the framework cannot guess the previous step accurately in all cases. There is no default at this moment because Stepper doesn’t assume linearity and the steps array is not rendered like A -> B -> C. So going from C -> B is not guaranteed to be the right default.

For example, in the onboarding flow the domains step is followed by either the plans step, or the use-my-domain step, then by the plans step. So, if you’re in the plans step and click back, Stepper can’t tell if you were at the domains step or use-my-domain step. But the flow author should know. Nonetheless, in it is guessable in most cases and I made this issue earlier to address it: https://github.com/Automattic/wp-calypso/issues/101509